go/datas/local_batch_store.go

```diff
@@ -11,53 +11,40 @@ import (
 	"github.com/attic-labs/noms/go/constants"
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
-	"github.com/attic-labs/noms/go/nbs"
 	"github.com/attic-labs/noms/go/types"
 )
 
 type localBatchStore struct {
-	cs            chunks.ChunkStore
-	unwrittenPuts *nbs.NomsBlockCache
-	vbs           *types.ValidatingBatchingSink
-	once          sync.Once
+	cs   chunks.ChunkStore
+	once sync.Once
+	mu   sync.Mutex
+	cc   *completenessChecker
 }
 
 func newLocalBatchStore(cs chunks.ChunkStore) *localBatchStore {
-	return &localBatchStore{
-		cs:            cs,
-		unwrittenPuts: nbs.NewCache(),
-		vbs:           types.NewCompletenessCheckingBatchingSink(cs),
-	}
+	return &localBatchStore{cs: cs, cc: newCompletenessChecker()}
 }
 
 // Get checks the internal Chunk cache, proxying to the backing ChunkStore if
 // not present.
 func (lbs *localBatchStore) Get(h hash.Hash) chunks.Chunk {
 	lbs.once.Do(lbs.expectVersion)
-	if pending := lbs.unwrittenPuts.Get(h); !pending.IsEmpty() {
-		return pending
-	}
 	return lbs.cs.Get(h)
 }
 
 func (lbs *localBatchStore) GetMany(hashes hash.HashSet, foundChunks chan *chunks.Chunk) {
-	remaining := make(hash.HashSet, len(hashes))
-	for h := range hashes {
-		remaining.Insert(h)
-	}
-	localChunks := make(chan *chunks.Chunk)
-	go func() { defer close(localChunks); lbs.unwrittenPuts.GetMany(hashes, localChunks) }()
-	for c := range localChunks {
-		remaining.Remove(c.Hash())
-		foundChunks <- c
-	}
-	lbs.cs.GetMany(remaining, foundChunks)
+	lbs.cs.GetMany(hashes, foundChunks)
 }
 
-// SchedulePut simply calls Put on the underlying ChunkStore.
+// SchedulePut calls Put on the underlying ChunkStore and adds any refs in c
+// to a pool of unresolved refs which are validated against the underlying
+// ChunkStore during Flush() or UpdateRoot().
 func (lbs *localBatchStore) SchedulePut(c chunks.Chunk) {
 	lbs.once.Do(lbs.expectVersion)
-	lbs.unwrittenPuts.Insert(c)
+	lbs.cs.Put(c)
+	lbs.mu.Lock()
```

Comment:
if you moved it outside because the server doesn't need to worry about concurrency, FWIW, I suspect that given the other work that's happening there, the mutex around each chunk isn't significant.